### ISSUE TITLE
space.createTeamMemebership typing is incorrect

### ISSUE THREAD

**Issue by ishabo** [contentful/contentful-management.js#388]
Hi,

I am getting the following error when I try to run `space.createTeamMemebership` due to a mismatch in the expected typing of the second param data:

```bash
UnknownKey: {
  "status": 400,
  "statusText": "Bad Request",
  "message": "The body you sent contains an unknown key.",
  "details": {
    "errors": [
      {
        "keys": [
          "id",
          "type",
          "link_type"
        ]
      }
    ]
  },
  "request": {
    "url": "team_space_memberships",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/vnd.contentful.management.v1+json",
      "X-Contentful-User-Agent": "sdk contentful-management.js/5.27.1; platform node.js/v12.4.0; os macOS/19.4.0;",
      "Authorization": "Bearer ...ucwDg",
      "user-agent": "node.js/v12.4.0",
      "Accept-Encoding": "gzip",
      "x-contentful-team": "__REPLACED_TEAM_ID__",
      "Content-Length": 89
    },
    "method": "post",
    "payloadData": "{\"admin\":false,\"roles\":[{\"id\":\"__REPLACED_ROLE_ID__\",\"type\":\"Link\",\"linkType\":\"Role\"}]}"
  },
  "requestId": "9ca48952ce0ff0bfb0e00c77a04c6a1b"
}
    at errorHandler (/Users/Shared/Dev/MMT/platform-cms/node_modules/contentful-management/dist/contentful-management.node.js:12996:17)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
    at async /Users/Shared/Dev/MMT/platform-cms/packages/scripts/cms-config-processor/out/dist/teams/helpers/createTeamsMemeberships/index.js:17:3
    at async Promise.all (index 0)
    at async /Users/Shared/Dev/MMT/platform-cms/packages/scripts/cms-config-processor/out/dist/helpers/processAll/index.js:16:3
    at async Promise.all (index 1)
    at async /Users/Shared/Dev/MMT/platform-cms/packages/scripts/cms-config-processor/out/dist/index.js:46:5
    at async Promise.all (index 0) {
  name: 'UnknownKey',
  message: '{\n  "status": 400,\n  "statusText": "Bad Request",\n  "message": "The ' +
    'body you sent contains an unknown key.",\n  "details": {\n    "errors": ' +
    '[\n      {\n        "keys": [\n          "id",\n          "type",\n        ' +
    '  "link_type"\n        ]\n      }\n    ]\n  },\n  "request": {\n    "url": ' +
    '"team_space_memberships",\n    "headers": {\n      "Accept": ' +
    '"application/json, text/plain, */*",\n      "Content-Type": ' +
    '"application/vnd.contentful.management.v1+json",\n      ' +
    '"X-Contentful-User-Agent": "sdk contentful-management.js/5.27.1; ' +
    'platform node.js/v12.4.0; os macOS/19.4.0;",\n      "Authorization": ' +
    '"Bearer ...ucwDg",\n      "user-agent": "node.js/v12.4.0",\n      ' +
    '"Accept-Encoding": "gzip",\n      "x-contentful-team": ' +
    '"__REPLACED_TEAM_ID__",\n      "Content-Length": 89\n    },\n    ' +
    '"method": "post",\n    "payloadData": ' +
    '"{\\"admin\\":false,\\"roles\\":[{\\"id\\":\\"__REPLACED_ROLE_ID__\\",\\"type\\":\\"Link\\",\\"linkType\\":\\"Role\\"}]}"\n' +
    '  },\n  "requestId": "9ca48952ce0ff0bfb0e00c77a04c6a1b"\n}'
}

```

I am calling `space.createTeamSpaceMembership` with the following params:

```ts
        const roles = ['roleName1', 'roleName2']
        await createTeamMemebership(teamId!, {
          admin: false,
          roles: roles.map(role => {
            return {
              id: role,
              type: 'Link',
              linkType: 'Role',
            }
          }),
        })
```

Which conforms to the typing of `createTeamSpaceMembership` as defined in `contentful-management`:

```ts
    createTeamSpaceMembership(teamId: string, data: Omit<TeamSpaceMembershipProps, 'sys'>): Promise<import("./entities/team-space-membership").TeamSpaceMembership>;
```

Where `TeamSpaceMembership` is defined as:

```ts
export declare type TeamSpaceMembershipProps = {
    sys: MetaSysProps;
    admin: boolean;
    roles: MetaLinkProps[];
};

```
And `MetaLinkProps` defined as 

```ts
export interface MetaLinkProps {
    type: string;
    linkType: string;
    id: string;
}
```

The way I fixed it is by changing the the roles data object that `createTeamSpaceMembership` to be included inside a `sys` key like this:

```ts
        const roles = ['roleName1', 'roleName2']
        await createTeamMemebership(teamId!, {
          admin: false,
          roles: roles.map(role => {
            const roleSysProps = {
              id: role,
              type: 'Link',
              linkType: 'Role',
            }
            return ({
              sys: roleSysProps,
            } as unknown) as MetaLinkProps
          }),
        })
```

In other words, I had to change:

```
{
  id: role,
  type: 'Link',
  linkType: 'Role',
}
```
to 

```
{
  sys: {
    id: role,
    type: 'Link',
    linkType: 'Role',
  }
}
```
And as you can see I have to force cast this which isn't great. Can someone confirm if this is indeed an issue?

Thanks in advance.

**Comment by vbeleuta** [comment]
Hello @ishabo,

Thank you for pointing this out to us. This was indeed an issue in our implementation and the call should be

`const roles = ['roleName1', 'roleName2']
        await createTeamSpaceMembership(teamId!, {
          admin: false,
          roles: roles.map(role => {
            return { 
              sys: {
                id: role,
                type: 'Link',
                linkType: 'Role',
             }
           }
          }),
        })`

I pushed a fix and is available in release 6.0.3. Let me know if you still have any issues.
